### PR TITLE
Interoperability tests target, protos and generated code

### PR DESCRIPTION
**Pull request by glbrntt**
- Adds interoperability test target, protos for the testing service and generated code

### CODE REVIEW

File: Sources/SwiftGRPCNIOInteroperabilityTests/protos/generate.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+protoc "src/proto/grpc/testing/messages.proto" \
+  --plugin=${PLUGIN_SWIFT} \
+  --plugin=${PLUGIN_SWIFTGRPC} \
+  --swift_out=${OUTPUT} \
+  --swift_opt=FileNaming=${FILE_NAMING},Visibility=${VISIBILITY} \
+  --swiftgrpc_out=${OUTPUT} \
+  --swiftgrpc_opt=NIO=true,FileNaming=${FILE_NAMING},Visibility=${VISIBILITY}
+
+echo "The following modifications must be made to the generated 'Grpc_Testing_TestServiceProvider_NIO' protocol code in ${OUTPUT}/test.grpc.swift:"
```
Comment (MrMage) [inline Sources/SwiftGRPCNIOInteroperabilityTests/protos/generate.sh]: Would you mind adding an explainer for why these changes need to be done?

File: Sources/SwiftGRPCNIOInteroperabilityTests/protos/src/proto/grpc/testing/empty.proto [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+// An empty message that you can re-use to avoid defining duplicated empty
+// messages in your project. A typical example is to use it as argument or the
+// return value of a service API. For instance:
+//
+//   service Foo {
+//     rpc Bar (grpc.testing.Empty) returns (grpc.testing.Empty) { };
+//   };
+//
+message Empty {}
```
Comment (MrMage) [inline Sources/SwiftGRPCNIOInteroperabilityTests/protos/src/proto/grpc/testing/empty.proto]: Could you use `SwiftProtobuf.Google_Protobuf_Empty` instead? (Let me know if this all has just been copied over from the "official" interop tests.)

Comment (glbrntt) [reply Sources/SwiftGRPCNIOInteroperabilityTests/protos/src/proto/grpc/testing/empty.proto]: These are copied from the official interop tests, the package naming also impacts the name of the generated type here, i.e. this becomes `Grpc_Testing_Empty` in Swift.
